Shuttle (New): WSE 'Nook'

Chervjakys opened the pull request:
<!-- Guidelines: https://docs.spacestation14.io/en/getting-started/pr-guideline -->

## About the PR
Adds the 'Nook' shuttle, functionally similar to the Camper.

## Why / Balance
While it is similar to the Camper, Nook provides more space for modifications done by the players (cargo bay, bedroom and restroom, main hall), has more thrusters to provide better mobility (with a uranium generator to power them), and a dedicated EVA airlock in front of the ship.

## How to test
Buy it, fly it, use it, sell it.

## Media
![ugolFinFin2](https://github.com/user-attachments/assets/288a336b-c39d-4b30-95ce-3f9b97b33d93)
![ugolFinFin](https://github.com/user-attachments/assets/5b721b5b-31a7-4a9d-99ab-34249429d523)

## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->


**Changelog**
<!-- Add a Changelog entry to make players aware of new features or changes that could affect gameplay.
Make sure to read the guidelines and take this Changelog template out of the comment block in order for it to show up.
Changelog must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog. -->

:cl: 
- add: Added Nook, a civilian shuttle.

Chervjakys commented:
Thank you for taking a look at my ship, Dusty.
 I tried working on things you've pointed out:
 
![ugol_v2_1](https://github.com/user-attachments/assets/ae28cc48-3bfd-4066-9a65-0c712005d863)
![ugol_v2_2](https://github.com/user-attachments/assets/52b4919d-b477-4fae-8e72-e2a3fc18a9dc)

> I feel like it might be nice to move the shuttle console to the middle, but that's a personal preference.

The railing setup you suggested feels a little bit nicer. The pilot's seat will stay in the same place for better vision (I replaced the wall with the button with a shuttle window to aid it)

> I would add a frame around your shutter button.

I won't be adding frames to buttons other than the one in the cockpit since they are not removable by the player (for some reason)

> Air sensors are very good for addressing changes in large spaces. ... You can probably cut down your air alarms a bit.

I've reduced the amount of air alarms to 2 (one for engineering and cargo bay, one for the rest of the ship, though one for the entire ship might be enough (?)), fire alarms to 3, removed the air sensors (vents and scrubbers sense air too, who would've thought!).

> ...you might get a little more shape variety and not lose much from a practical standpoint by dropping the two side small thrusters.

After comparing the ship with and without additional side thrusters, the difference in handling between them was minimal, so I removed them.

> The EVA locker by the egress bay does like to scoop up toolboxes.

There's now an additional locker in the frontal airlock for storing the toolboxes, I also relocated the substation closer to the center of the ship.

> However I feel very strongly that you'll want to replace your airlocks leading outside with the glass variety.

Replaced the airlocks at the cargo bay with glass variants and shifted the light's locations. Not sure if I like them transparent though.

Chervjakys commented:
I'll redo the change on the airlocks in the next update,
  just to explain the placement of the railing: I put them there specifically to prevent connection induced rubber-banding forcing you out of the shuttle once you hop out of your seat. It's fairly easy for the player to remove the railing if they don't like it (there are even tools that come with the shuttle to do just that!), so I'll keep it the way it is.
 Not sure what I can do about the fire alarms.
 The window button will go to the wall for sure! (probably)
 I like how the hull plating looks under thrusters on the ship right now, I'll replace it with lattice if I'll get asked to do so by the maintainers.

arimah reviewed (dismissed):
Hello!

I want to start this review by saying I have an enormous soft spot for ships like this, that exist purely for relaxation and not for profitmaxing. As you read the following critique, assume I say everything from a position of love and admiration.

I'm going to focus on the overall layout and design of the ship, as I believe it will require some larger changes before it's truly ready for Frontier. :)

First a technical note: you _must_ rename the grid file to `ugol.yml`, with a lowercase u. All the other grid files start with a lowercase letter. The tests keep breaking because of this (some file systems are case-sensitive).

---
**The name**
I'm gonna be honest, "Ugol" is kind of, well, an ugly name. I don't know if the word has any other meanings, but Ugol is _very_ close to "ugly". :( This is not a dealbreaker, but I would ask you to think really hard about potential other names for this nugget.

---
**"Thruster balconies"**
![image](https://github.com/user-attachments/assets/9fde5809-f4ae-4ab0-9abd-6e8d026f6ca4)

I don't know who started this trend, and I don't know why it persists. If you want to keep the shape, please remove the airlocks and extend the walls so the thrusters are flush with the ship. These thruster balconies serve _no other purpose_ than to give you access to the thrusters, and they make the thrusters look improperly anchored to the ship. Consider that these are big machines that generate enormous amounts of thrust.

Also, lattice under your thrusters, not plating. :)

---
**Bed and bath**
![image](https://github.com/user-attachments/assets/710d2b3b-d29e-4106-96c4-bb4f83221919)

The bathroom is very cute. Minimalistic, but nice. However, that mirror is actually on the outside of the ship; rotate it 180°, please. Also the use of a concrete trim on a non-concrete floor is _weird_. Please experiment with other quarter-tile decals if you want a trim. Alternatively, it might actually look nice to omit the trim altogether. Sometimes less is more.

The bedroom, however. Due to the placement of the doors, you effectively have only those two usable tiles. There's nowhere else to place furniture without "blocking" (visually, if not mechanically). Consider this hypothetical door layout, which gives you _four tiles_ to place stuff on:
![image](https://github.com/user-attachments/assets/ad4b4f73-847a-49e7-bd3f-872cfb2aa3da)

e.g.:
![image](https://github.com/user-attachments/assets/62edc2fa-30bb-410c-8ed1-cdf05bb9322f)
Now it looks so much more like a real, usable bedroom :) Needs some finishing touches, but I hope you get the idea. The bathroom would need to be relayouted for this to work, but not hugely.

The placement of the shutter button is also not ideal. Move it to a location that wouldn't require you to climb over the furniture to reach it.

---
**Engineering and EVA access**
![image](https://github.com/user-attachments/assets/655d6af7-443a-4eb4-a393-1f88bbd1db7a)
Engineering is decent! Does what it needs to. Decal work is minimal and pleasant.

The EVA access thing is awkward. One single tile to stand on, and _two_ wallmount lockers that open up onto the same rack, which already has a bunch of stuff on it.

If you were to remove the thruster balconies (as opposed to just shifting the thrusters up), you can extend engineering downwards a bit and give the EVA access more room to breathe. As an example:
![image](https://github.com/user-attachments/assets/640a986e-2c35-4cec-acf4-e56a19a57dee)
Obviously you'd have to rework all the decals and floor textures and stuff. I'm not sure the ship needs _two_ toolboxes with almost the same tools, but everything fits and nothing gets in the way. You could even make this the only way into engineering, so the captain doesn't have to pass through the entryway/airlock area.

---
**Main lounge and bridge**
![image](https://github.com/user-attachments/assets/6878bf04-8349-4715-b074-480578e6434a)

I'm gonna be honest, that lounge does not feel like a lounge at all. Sofa phasing into the wall, facing a table with a TV... It's not _quite_ enough to sell the idea to me. That TV should not be mounted on a table, and the paper bin looks really cramped with the potted plant.

The bridge is pretty minimal, but personally I find the railing restricts the already super constrained space. Also the window with the button should be a wall. Sorry, no window-mounted buttons. :)

Taking the other suggestions into account, consider how the ship might work with the sofa turned around so it looks like it's actually facing the TV, and with an extended fax table functioning as a natural barrier/separator between the lounge and the bridge:
![image](https://github.com/user-attachments/assets/cda66f15-9fcd-4313-ab45-08a1e4ef2a38)
I think it's worth experimenting with.

And again, not a huge fan of concrete trims on non-concrete flooring. It looks a little odd.

---
**Entryway**
![image](https://github.com/user-attachments/assets/6ddd4293-720c-4adb-9e99-8ee9b660f836)
The combination of store room and entryway airlock is fine in my book. The railing makes the passage into the rest of the ship pretty narrow and a little fiddly to navigate, however.

Taking some of my other suggestions to heart, consider how the ship might look if you were to move the inner wall down by one tile and change the storage over to the other side:
![image](https://github.com/user-attachments/assets/7fd948c3-dbe3-4a20-8988-8891f162671f)

Obviously the wall-mounted things would have to be moved and such, but... think about it and play around a bit. :)

---
**Overall impressions**
Man, this ship is _dirty_. For the price point, it's probably okay, but don't go completely overboard. Several of the dirt decals are between tiles; please snap them to the tile instead.

I like the size and shape. It's a very cute addition to the lineup of personal ships. With some more work on the layout, I think we can accept it... but _maybe_ under a new name. :)

Thank you for your contribution. I look forward to seeing this develop.

arimah commented:
Oh, a few other notes I meant to include but didn't:

- One air alarm is sufficient for a ship this size
- Fire alarms are unnecessary

Chervjakys commented:
Helloooo!

> Ugol is very close to "ugly"

'Ugol' translated would mean "angle", I did not intend for people to interpret it this way XD
I don't really have any ideas for a better name myself, if there even could be one.

### Thruster balconies

I am unsure as to why it's a bad thing, but they are not crucial, so those can go!

### Bedroom and bathroom layout

Thank you for reminding me that I could simply move the airlocks around, now the bedroom should feel a bit more furnished.
I think I chose the concrete trim just because it had more texture to it. But yeah, it doesn't really match the floor tiles due to it's odd coloring.
_By the way, I have no idea why, but the bucket item just seems to... disappear whenever I buy the shuttle._

### EVA access

Moved the wall down, now there should be a bit more tiles than just one to stand on. I won't be adding an airlock leading into this compartment because the door won't feel right in the lounge. I'd much rather have it lead to engineering and then to the cargo bay, so that the 'work area' is contained and away from the living zone.

> I'm not sure the ship needs two toolboxes with almost the same tools

Unfortunately, they are **_almost_** the same tools!

### Lounge and bridge

I tried to put a wall-mounted TV on the right side, moving the couch to the middle of the room and keeping the table between it and the cockpit, but it did not sit right with me so I just rotated the sofa around.

Moved the button to the left side to keep the window so the view is not as restricted.

### Cargo bay

Dusty suggested turning the storage area around too, but:
> just to explain the placement of the railing: I put them there specifically to prevent connection induced rubber-banding forcing you out of the shuttle once you hop out of your seat. It's fairly easy for the player to remove the railing if they don't like it (there are even tools that come with the shuttle to do just that!), so I'll keep it the way it is.

If I lower the wall down, then the lounge will turn into a hallway, which is not a good thing if it's connecting two work areas: not very feng-shui!

### Misc
I tried to put a wall-mounted TV on the right side, moving the couch to the middle of the room and keeping the table between it and the cockpit, but it did not sit right with me so I just rotated the sofa around.

Moved the button to the left side to keep the window so the view is not as restricted.

### Cargo bay

Dusty suggested turning the storage area around too, but:
> just to explain the placement of the railing: I put them there specifically to prevent connection induced rubber-banding forcing you out of the shuttle once you hop out of your seat. It's fairly easy for the player to remove the railing if they don't like it (there are even tools that come with the shuttle to do just that!), so I'll keep it the way it is.

If I lower the wall down, then the lounge will turn into a hallway, which is not a good thing if it's connecting two work areas: not very feng-shui!

### Misc

> Man, this ship is dirty.

Yes!

![ugolFinFin](https://github.com/user-attachments/assets/fe4f5bc3-6205-4585-8078-6a2beb4266f1)
![ugolFinFin2](https://github.com/user-attachments/assets/bb00f9eb-9de1-44b0-a974-d762c0e6bc0d)

Now there's only one air alarm and I removed the fire alarms as well

> Man, this ship is dirty.

Yes!

![ugolFinFin](https://github.com/user-attachments/assets/fe4f5bc3-6205-4585-8078-6a2beb4266f1)
![ugolFinFin2](https://github.com/user-attachments/assets/bb00f9eb-9de1-44b0-a974-d762c0e6bc0d)

Now there's only one air alarm and I removed the fire alarms as well

arimah commented:
That is a marked improvement that still retains the spirit of the ship. Very, very good! I shall review it properly in the next day or so. :D

arimah commented:
I solved the Mystery of the Vanishing Bucket:
![image](https://github.com/user-attachments/assets/9190f66d-7a67-4522-8efe-a4746dd6aedc)

... it gets picked up by the trash bag magnet. The trash bagnet. We may need to revisit the whitelist there, but separate PR. You can keep it as-is for now.

CloudOfNeutronFlux commented:
>'Ugol' translated would mean "angle", I did not intend for people to interpret it this way XD
I don't really have any ideas for a better name myself, if there even could be one.

You could name it Nook, which 'ugol' can be interpreted as (even if, technically speaking, 'nook' directly translates to the diminutive form, 'ugolok'). The vibes definitely match.

Chervjakys commented:
> You could name it Nook, which 'ugol' can be interpreted as (even if, technically speaking, 'nook' directly translates to the diminutive form, 'ugolok'). The vibes definitely match.

The vibes do match! That's a pretty good suggestion, I'll keep it in mind

Chervjakys commented:
should've done this earlier but the shuttle is now renamed

Cheackraze commented:
looks like network configurator bugged out on ship save. You'll have to load up and re-save the ship file, and possibly re-link any badly linked devices if it still fails

Chervjakys commented:
Whoops I might've done something

arimah commented:
@Chervjakys Did you intend to close this? I'm happy to reopen it for you.

Chervjakys commented:
No, not at all, I did this by accident

Chervjakys commented:
> You'll have to load up and re-save the ship file, and possibly re-link any badly linked devices if it still fails
did both just to be sure

Chervjakys commented:
should be all done! im still not sure about that failing workflow thing though

whatston3 commented on Resources/Prototypes/_NF/Shipyard/nook.yml:
```diff
@@ -0,0 +1,44 @@
+            prefixCreator: '14'
+        - type: StationJobs
+          availableJobs:
+            Contractor: [ 0, 0 ]
+            Pilot: [ 0, 0 ]
+            Mercenary: [ 0, 0 ]
```
```suggestion
            ContractorInterview: [ 0, 0 ]
            PilotInterview: [ 0, 0 ]
            MercenaryInterview: [ 0, 0 ]
```
Use new interview jobs.
